src/main/java/uk/nhs/hee/tis/trainee/ndw/service/DataLakeFacade.java

```diff
@@ -0,0 +1,71 @@
+package uk.nhs.hee.tis.trainee.ndw.service;
+
+import com.azure.storage.file.datalake.DataLakeDirectoryClient;
+import com.azure.storage.file.datalake.DataLakeFileSystemClient;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class DataLakeFacade {
+
+  private final DataLakeFileSystemClient dataLakeClient;
+
+  DataLakeFacade(DataLakeFileSystemClient dataLakeClient) {
+    this.dataLakeClient = dataLakeClient;
+  }
+
+  /**
+   * Create the required year-month-day subdirectories based on current date.
+   *
+   * @return The directory client for the day subdirectory.
+   */
+  public DataLakeDirectoryClient createYearMonthDaySubDirectories(
+      DataLakeDirectoryClient root) {
+    Instant now = Instant.now();
+    ZoneId utcZone = ZoneId.of("UTC");
+
+    return root
+        .createSubdirectoryIfNotExists(
+            DateTimeFormatter.ofPattern("'year='yyyy").withZone(utcZone).format(now))
+        .createSubdirectoryIfNotExists(
+            DateTimeFormatter.ofPattern("'month='yyyyMM").withZone(utcZone).format(now))
+        .createSubdirectoryIfNotExists(
+            DateTimeFormatter.ofPattern("'day='yyyyMMdd").withZone(utcZone).format(now));
+  }
+
+  /**
+   * Create the required subdirectory.
+   *
+   * @return The directory client for the required subdirectory.
+   */
+  public DataLakeDirectoryClient createSubDirectory(String root, String name) {
```

Comment:
I don't super love the passing of a DataLakeDirectoryClient back each time, it feels clumsy. I had a look at writing this as an extension class of the DataLakeFileSystemClient, but that has a package-private constructor, and it felt like angels-fear-to-tread territory.